File: Scripty/Library/Shell.swift
//
//  Shell.swift
//  Scripty
//
//  Created by Ivan Smetanin on 18/03/2018.
//  Copyright © 2018 Ivan Smetanin. All rights reserved.
//

import Foundation

final class Shell {

    @discardableResult
    static func run(_ arguments: [String]) -> Int32 {
        debugPrint("Task \(arguments) is running")
        let task = Process()
        task.launchPath = "/usr/bin/env"
        task.arguments = arguments
        task.launch()
        task.waitUntilExit()
        debugPrint("Result status: \(task.terminationStatus)")
        return task.terminationStatus
    }

}

File: Scripty/Library/Utils/Shell.cpp
//
//  Shell.cpp
//  Scripty
//
//  Created by i.smetanin on 01.09.2020.
//  Copyright © 2020 Ivan Smetanin. All rights reserved.
//

#include "Shell.h"
#include <iostream>
using namespace std;

__API_AVAILABLE(macos(10.0)) __IOS_PROHIBITED
__WATCHOS_PROHIBITED __TVOS_PROHIBITED
void shellCMD_V(const char *cmd) {
    system(cmd);
}

__API_AVAILABLE(macos(10.0)) __IOS_PROHIBITED
__WATCHOS_PROHIBITED __TVOS_PROHIBITED
int shellCMD_I(const char *cmd) {
    int out = system(cmd);
    return WEXITSTATUS(out);
}
